### PR TITLE
Major update to Tanker

### PR DESCRIPTION
**Pull request by JackDanger**
Ahoy Kidpollo!

Thanks for the fantastic work with Tanker! I really love it, it saved me weeks of work.

I've been using Sunspot and Solr for a year and I just switched to IndexTank. I tried ThinkingTank but it was nowhere near production-ready. Instead I forked Tanker and started adding all the functionality I've grown accustomed to from other search ORM extensions.

What I've added to Tanker:

block-style field definitions:

```
tankit 'index' do
  indexes :author_names do
    authors.map(&:name)
  end
end
```

document variable definitions

```
tankit 'index' do
  variables do
    {0 => id,
     1 => lat,
     2 => lng}
  end
end
```

Multi-class searching:

```
Tanker.search([Topic,User,Post], 'keyword') # returns mixed results of different types
```

Single-class reindexing:

```
Topic.tanker_reindex
```

Oh, and batch indexing. When you run 'rake tanker:reindex' it'll now default to indexing and PUTting 200 records at a time. WAY, WAY, faster than doing it one at a time. I'm indexing 800 records in about 15 seconds.

Automatic timestamping: if the object responds to 'created_at' then that'll be set as the document's creation time in the index.

I also cleaned up the 'tankit' configuration call so it doesn't pollute class-level and object-level methods. I put it into a ModelConfig class so users don't have their Topic.index, Topic.indexes, etc. methods overwritten.

I've added specs for everything and did my best to preserve all of your original organization. I didn't mess with anything that I didn't have to. I've also updated the README.

Care to push a new gem version? :)

### CODE REVIEW

**Comment by kidpollo** [comment]
Hey JAck!

I greatly appreciate your input, you have no idea how happy I get when I get a pull request! I have been kinda busy and I feel bad I have not integrated your stuff and am happy to continue colaborating with you on this awesome gem :P

I am thingking on promoting this to heroku directly I moved to san francisco and I have some friends there :P

I have not seen the code but did you write specs around your features? If not then it would be cool to have them.

Thanks

PAco

**Comment by JackDanger** [comment]
> I am thingking on promoting this to heroku directly I moved to san
> francisco and I have some friends there :P

Awesome :)

> I have not seen the code but did you write specs around your features? If
> not then it would be cool to have them.

Yup, the only thing that isn't specced is a Rake task for updating
server-side function definitions. But it wouldn't be too hard to add one.
Otherwise it's all tested and working great.

Thanks for doing the hard 90% of the work, it was fun to just add features
on top of a good code base. I started using index_tank and then
thinking_tank and couldn't bear the awful code quality. Tanker was a joyful
find :)

::Jack

**Comment by JackDanger** [comment]
I've added support for filter_functions and filter_docvars (so you can now filter your results to just X miles from some point) and I've updated the README with an example of a fulltext geospatial search. The new code is working, tested, and live on my company site at http://cloops.com/search

File: README.rdoc [hunk 1]
```diff
@@ -50,7 +49,32 @@ in the IndexTank Dashboard
       tankit 'my_index' do
         indexes :title
         indexes :content
+        indexes :id
         indexes :tag_list #NOTICE this is an array of Tags! Awesome!
+
+        # you may also dynamically retrieve field data
+        indexes :author_names do
```
Comment (kidpollo) [inline README.rdoc:56]: this is awesome!

**Pull request merged**